api/hearings/data/hearing-actuals.mock.data.ts

```diff
@@ -0,0 +1,123 @@
+    });
+  }
+  return hearingActuals;
+}
+
+function generateParties(numberOfParties: number): Party[] {
+  const parties: Party[] = [];
+
+  for (let i = 0; i < numberOfParties; i++) {
+    const id = getRandomElement([1, 2, 3, 4, 5]);
+    parties.push({
+      individualDetails: {
+        title: getRandomElement(['Mr', 'Mrs', 'Miss', 'Ms']),
+        firstName: getRandomElement(['John', 'Jane', 'Mary', 'Bob', 'Tom']),
+        lastName: getRandomElement(['Smith', 'Jones', 'Brown', 'Smith', 'Jones']),
+      },
+      organisationDetails: {
+        cftOrganisationID: getRandomElement(['12345', '54321', '67890']),
+        name: getRandomElement(['Company A', 'Company B', 'Company C', 'Company D', 'Company E']),
+      },
+      partyId: id,
+      partyRole: id,
+      partyChannelSubType: getRandomElement(['Email', 'Phone', 'Fax', 'Letter', 'Other']),
+    });
+  }
+  return parties;
+}
+
+function generateActualDayParties(numberOfDayParties: number): ActualDayParty[] {
+  const dayParties: ActualDayParty[] = [];
+
+  for (let i = 0; i < numberOfDayParties; i++) {
+    const id = getRandomElement([1, 2, 3, 4, 5]);
+    dayParties.push({
+      actualPartyId: id,
+      actualIndividualDetails: {
+        firstName: getRandomElement(['John', 'Jane', 'Mary', 'Bob', 'Tom']),
+        lastName: getRandomElement(['Smith', 'Jones', 'Brown', 'Smith', 'Jones']),
+      },
+      actualOrganisationDetails: {
+        name: getRandomElement(['Company A', 'Company B', 'Company C', 'Company D', 'Company E']),
+      },
+      didNotAttendFlag: getRandomElement([true, false]),
+      partyId: id,
+      partyChannelSubType: getRandomElement(['Email', 'Phone', 'Fax', 'Letter', 'Other']),
+      partyRole: getRandomElement([1, 2, 3, 4, 5]),
+      representedParty: getRandomElement([1, 2, 3, 4, 5]),
+    });
+  }
+  return dayParties;
+}
+
+function generateActualHearingDays(numberOfActualHearingDays: number): ActualHearingDay[] {
+
+  const actualHearingDays: ActualHearingDay[] = [];
+
+  for (let i = 0; i < numberOfActualHearingDays; i++) {
+
+    const actualHearingDay: ActualHearingDay = {
+      hearingDate: getRandomElement(['2021-03-12', '2021-03-13', '2021-03-14']),
+      hearingStartTime: '2021-03-12T09:00:00.000+0000',
+      hearingEndTime: '2021-03-13T10:00:00.000+0000',
+      pauseDateTimes: [],
+      actualDayParties: generateActualDayParties(1),
+    };
+
+    actualHearingDays.push(actualHearingDay);
+  }
+  return actualHearingDays;
+}
+
+function generatePlannedHearings(numberOfPlannedHearingDays: number): PlannedHearing[] {
+  const plannedHearings: PlannedHearing[] = [];
+
+  for (let i = 0; i < numberOfPlannedHearingDays; i++) {
+    const plannedHearing: PlannedHearing = {
+      plannedStartTime: '2021-03-12T09:00:00.000+0000',
+      plannedEndTime: '2021-03-13T10:00:00.000+0000',
+      parties: generateParties(2),
+    };
+    plannedHearings.push(plannedHearing);
+  }
+
+  return plannedHearings;
+}
```

Comment:
Apprieciated for generating random value for a key in the models, but this has been proved in other projects like NoC/share case etc., it doesn't quite work well for demoing LIVE alike data, it is hard to maintain on development, not actually use the benefits for typescript models check, for those timings in requirement it can't be past time but here is random time including past, and also it can't reuse those reference data like hearing type, hearing result which they are from enum types etc. I would prefer this is just a hard coded mock data with using typescript type check on the models, just like we have done for get hearing details.

Reply:
Hmm I am not sure how it's hard to maintain. Probably the easiest way to create alot of fake data. About using typescript model checks I am not sure what you mean there. I am using the actual enums to generate random values. And the point about times and dates you could always create a helper function which creates times within a specific range.

Reply:
We used the existing faker plugin before to generate random value in mock, that proved it doesn't quite work and doesn't match the prototype we are provided, then we removed those 'Random' value logic, what we are trying to achieve here is simple, just create the hard code copy data from the prototype, from here https://ucd-xui2-prototype.herokuapp.com/hearing-actuals/version3/actuals-categories. In one of this scenario you have to control those start time < pause time < resume time < end time, and there are much more reference data you could see coming from the prototyp. It's not necessary to spend so much effort on this 'Random' value in mock data.

Reply:
Sure I will change it. I have all these helper functions already fyi about the comment  on effort.